File: packages/custos-ts/src/responders/noop.ts
// Noop responder — . Logs the prompt and auto-denies (for tests /
// headless). Mirrors `custos.responders.noop.NoopResponder`.

import type { Responder } from "./base.ts";
import type { PromptRequest, PromptResponse } from "../schema.ts";

export class NoopResponder implements Responder {
  readonly name = "noop";

  async prompt(_req: PromptRequest): Promise<PromptResponse> {
    // Log + auto-deny. : a responder down-fail (incl. `noop` here)
    // is treated as a safe `deny` — the  floor holds.
    return {
      choice: "deny",
      ttl: null,
      signature: null,
      nonce: null,
      approver: null,
    };
  }
}

File: src/custos/responders/noop.py
"""Noop responder: logs the prompt and auto-denies .

Intended for tests and headless runs where no human is present.
"""

from __future__ import annotations

from custos.responders.base import PromptRequest, PromptResponse
from custos.schema import Decision

__all__ = ["NoopResponder"]


class NoopResponder:
    name = "noop"

    def prompt(self, req: PromptRequest) -> PromptResponse:
        return PromptResponse(choice=Decision.DENY)
